VictoriaAlbert/VictoriaAlbert/MuseumObjectsTableViewController.swift

```diff
@@ -0,0 +1,127 @@
+            museumAPIEndPoint = museumAPI + searchKey
+        }
+        
+        
+        APIRequestManager.manager.getData(endPoint: museumAPIEndPoint) { (data: Data?) in
+            if  let validData = data,
+                let validObjects = MuseumObject.getMuseumObjects(from: validData) {
+                self.museumObjects = validObjects
+                
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }
+    }
+    
+    
+    // MARK: - Table view data source
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return museumObjects.count
+    }
+    
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "MuseumObjectCellIdentifier", for: indexPath)
+        
+        let museumObjectAtIndex = museumObjects[indexPath.row]
+        let ObjectDetailString = museumObjectAtIndex.objectName + ", " + museumObjectAtIndex.dateText+", " + museumObjectAtIndex.placeName
+        
+        cell.textLabel?.text = ObjectDetailString
+        cell.detailTextLabel?.text = museumObjectAtIndex.titleField
+        cell.imageView?.image = nil
+        
+        
+        
+        let imageId = museumObjectAtIndex.imageString
+        
+        if imageId != "" {
+            let imageIdArray = Array(imageId.characters)
+            var yearID = ""
+            
+            for i in 0...5 {
+                yearID += String(describing: imageIdArray[i])
+            }
+            
+            let imageEndURL = "http://media.vam.ac.uk/media/thira/collection_images/"
+            let slash = "/"
+            let sizeString = "_jpg_o"
+            let imageTypeString = ".jpg"
+            
+            
+            let imageEndPoint = imageEndURL + yearID + slash + imageId + sizeString + imageTypeString
+            
+            APIRequestManager.manager.getData(endPoint: imageEndPoint ) { (data: Data?) in
+                if  let validData = data,
+                    let validImage = UIImage(data: validData) {
+                    DispatchQueue.main.async {
+                        cell.imageView?.image = validImage
+                        cell.setNeedsLayout()
+                        
+                    }
+                }
+            }
+        }
+        else {
+            cell.imageView?.image = UIImage(named: "noPreview")
+        }
+        
+        return cell
+    }
+    
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```

Comment:
There is no need for both a segue and didSelectRowAt. The pattern we've been following in class and that I recommend sticking with is one segue and finding the selected object via the sender in prepare(for segue:). The sender is cast to a UITableView cell from which you can ask the table what it's index path is.The index path is used to find the data as with cellForRow.